src/subscript-loc.c

```diff
@@ -92,23 +101,43 @@ r_obj* lgl_as_location(r_obj* subscript,
   if (subscript_n == 1) {
     int elt = r_lgl_get(subscript, 0);
 
-    r_obj* out;
+    r_ssize recycle_size = n;
+
+    r_obj* out = r_null;
+    r_keep_loc out_shelter;
+    KEEP_HERE(out, &out_shelter);
+
     if (elt == r_globals.na_lgl) {
-      out = KEEP(r_alloc_integer(n));
-      r_int_fill(out, r_globals.na_int, n);
+      switch (opts->missing) {
+      case SUBSCRIPT_MISSING_PROPAGATE: {
+        out = r_alloc_integer(n);
+        KEEP_AT(out, out_shelter);
+        r_int_fill(out, r_globals.na_int, n);
+        break;
+      }
+      case SUBSCRIPT_MISSING_REMOVE: {
+        out = vctrs_shared_empty_int;
```

Comment:
i.e.

```r
vec_as_location(NA, n = 5, missing = "remove")
```

Reply:
I think we're going to assign names to that shared object, below?

Reply:
Good catch!